en/Search Algorithms/Floyd Cycle Detection Algorithm to Find Duplicate Number.md

```diff
@@ -0,0 +1,38 @@
+# Floyd Cycle Detection Algorithm to find duplicate number in an array
+
+## Problem Statement
+
+Given an array of integers containing `n + 1` integers, where each integer is in the range `[1, n]` inclusive. If there is only one duplicate number in the input array, this algorithm returns the duplicate number without modifying the original array, otherwise, it returns -1.
+
+## Approach
+
+- Use the function `f(x) = arr[x]` to construct the sequence: 
+`arr[0]`, `arr[arr[0]]`, `arr[arr[arr[0]]]`, `arr[arr[arr[arr[0]]]]`, etc....
+- Each new element in the sequence is an element in `arr[]` at the index of the previous element.
+- Starting from `x = arr[0]`, it will produce a linked list with a cycle.
+- The cycle appears because `arr[]` contains duplicate elements(at least one). The duplicate value is an entrance to the cycle. 
+
+## Time Complexity
+
+O(n)
```

Comment:
The comment in the linked C implementation reads:

> O(1) space and the time complexity is less than O(n^2)

That's not the same.

Reply:
What does the C implementation mean by `less than O(n^2)`? What's it in practice? Do they mean that it can be `O(n log n)` or `O(n)`? If so, can you provide an example of `O(n log n)`?

Reply:
I think this is just the constraint and it's not really related to the implementation of the algorithm itself. Based on this constraint, we are presenting a `O(n)` solution which is correct.